homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,218 @@
+"""Config flow for ezviz."""
+import logging
+
+from pyezviz import EzvizClient
+from pyezviz.client import PyEzvizError
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import (  # pylint: disable=unused-import
+    ATTR_CAMERAS,
+    ATTR_SERIAL,
+    CONF_FFMPEG_ARGUMENTS,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ezviz."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def _validate_and_create_auth(self, data):
+        """Validate the user input allows us to connect.
+
+        Data has the keys from data_schema with values provided by the user.
+        """
+        await self.async_set_unique_id(data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        # constructor does login call
+        client = EzvizClient(
+            data[CONF_USERNAME],
+            data[CONF_PASSWORD],
+            data.get(CONF_REGION, DEFAULT_REGION),
+            data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        # Validate data by sending a login request to ezviz
+
+        try:
+            await self.hass.async_add_executor_job(client.login)
+
+        except PyEzvizError as err:
+            raise InvalidAuth from err
+
+        auth_data = {
+            "username": data[CONF_USERNAME],
+            "password": data[CONF_PASSWORD],
+            "region": data.get(CONF_REGION, DEFAULT_REGION),
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    async def _create_camera_rstp(self, data):
+        """Create RSTP auth entry per camera in config."""
+
+        await self.async_set_unique_id(data[0])
+        self._abort_if_unique_id_configured()
+
+        camera_rstp_creds = {
+            ATTR_SERIAL: data[0],
```

Comment:
Please use `CONF_*` constants for all config keys.

Reply:
Maybe we don't need to store the serial in the data but can just use the entry.unique_id attribute?

Reply:
I'm passing the data via camera.py, cleaned up config_flow to pass data directly to async_create_entry.

The code is now simpler but I'm not sure if i completely resolved your request? I can't yet find a way to skip storing the camera serial as the init setup routines depend on this. The camera rtsp entities aren't "real" entities so to speak, and are used just for camera rtsp credentials.

Reply:
We set the serial as unique_id on the config entry in the config flow when configuring the camera(s). So when the camera platform is loaded we can get the serial via `entry.unique_id`.

Reply:
The cameras are loaded from the cloud (IP addresses and what cameras are registered). The camera entries are just used for the local rtsp stream to be able to see a camera feed. 

The camera entries also have a different ids than that of the ezviz account. I'm passing the rtsp username and password to the camera platform via hass.data[DOMAIN][entry.unique_id] and then matching them under camera "async_setup_entry"

If i save the camera rtsp credentials under the "ezviz cloud" entry then there is no way the user can add/remove credentials for new cameras. 

I'm struggling to find a better way of doing this. Creating the ezviz cloud entry as a bridge might be an alternative but this might not follow the standards all that well? (Ezviz cloud entry almost like a hue bridge?)